include/boost/date_time/gregorian/greg_duration.hpp

```diff
@@ -84,26 +84,46 @@ namespace gregorian {
       base_type::operator-= (rhs);
       return *this;
     }
+    
+    #if !defined(BOOST_EMBTC)
+      
     BOOST_CXX14_CONSTEXPR friend
```

Comment:
is the issue here that the compiler cannot inline this function for some reason?

Reply:
It can not find the inlined friend function for exported/imported classes unless the definition is placed outside the class itself. I have reported this to Embarcader with sample code at https://quality.embarcadero.com/browse/RSP-28133.

Reply:
Ok thanks. So maybe the better approach is to not do the macro switch and just do it the same way for all the compilers.  We could leave a comment about borland not supporting to explain why these guys aren't in the class itself.  Anyway, I'd like to minimize the amount of macros switches.  Can you refactor that way?

Reply:
and note you may have to resync the branch from remote because of other updates I resyncd some with develop

Reply:
Do you mean that you want me to change all similar to what I have done for the Embarcadero C++ clang-based compilers ? I can do that, and resync the PR to include the latest changes from 'develop'.

Reply:
yes -- the changes for Embarcadero should be valid on all the other compilers too -- might as well just do it one way for all of them without the #ifdef

Reply:
I already added the change in my PR.

Reply:
Thanks -- that looks much cleaner.